lib/Client.php

```diff
@@ -181,28 +210,146 @@ public function getFeatureFlag(
             "event" => '$feature_flag_called',
         ]);
 
-        if ($result) {
+        if (!is_null($result)) {
             return $result;
         }
         return $defaultValue;
     }
 
+    /**
+     * get the feature flag value for this distinct id.
+     *
+     * @param string $distinctId
+     * @param array $groups
+     * @param array $personProperties
+     * @param array $groupProperties
+     * @return array
+     * @throws Exception
+     */
+    public function GetAllFlags(
+        string $distinctId,
+        array $groups = array(),
+        array $personProperties = array(),
+        array $groupProperties = array()
+    ) : array
+    {
+        $response = [];
+        $fallbackToDecide = false;
+
+        if (count($this->featureFlags) > 0) 
+        {
+            foreach ($this->featureFlags as $flag) {
+                try {
+                    $response[$flag['key']] = $this->computeFlagLocally(
+                        $flag,
+                        $distinctId,
+                        $groups,
+                        $personProperties,
+                        $groupProperties
+                    );
+    
+                } catch (InconclusiveMatchException $e) {
+                    $fallbackToDecide = true;
+                }
+            }
+        } else {
+            $fallbackToDecide = true;
+        }
+
+        if ($fallbackToDecide) {
+            try {
+                $featureFlags = $this->fetchFeatureVariants($distinctId, $groups, $personProperties, $groupProperties);
+                $response = array_merge($response, $featureFlags);
+            } catch (Exception $e) {
+                // TODO: handle error
```

Comment:
debug logs here only?

Same for when successful, and we should add a way in the docs to 'enable' debug mode so they can see all these log messages. Off by default / WARN level by default so it doesn't spew logs during regular prod running